lib/v2/screens/profile_screens/citizenship/interactor/usecases/get_referred_accounts_use_case.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:async/async.dart';
+import 'package:seeds/v2/datasource/local/settings_storage.dart';
+import 'package:seeds/v2/datasource/remote/api/profile_repository.dart';
+import 'package:seeds/v2/datasource/remote/model/referred_accounts_model.dart';
+import 'package:seeds/v2/datasource/remote/model/profile_model.dart';
+
+class GetReferredAccountsUseCase {
+  final ProfileRepository _profileRepository = ProfileRepository();
+
+  Future<Result> run() async {
+    return _profileRepository.getReferredAccounts(settingsStorage.accountName).then((Result result) async {
+      if (result.isError) {
+        return result;
+      } else {
+        ReferredAccounts referredAccounts = result.asValue?.value as ReferredAccounts;
+        List<ProfileModel> accounts = [];
+        //This is an expensive approach we need change it
+        for (var i in referredAccounts.accounts) {
```

Comment:
this can be done a little better with 
Futures.wait(listOfFutures)

Reply:
look at the explore use case.

Reply:
Changed to Future.wait